### PR TITLE
add patch to fix memory leak in OpenMPI that heavily affects CP2K, SIESTA, QuantumESPRESSO, VASP, ...

### PR DESCRIPTION
**Pull request by hajgato**
There is a memory leak in `OpenMPI` `4.0.5`, `4.0.6`, `4.1.0`, and `4.1.1` which heavily affects `CP2K`, `SIESTA`, `QuantumESPRESSO`, `VASP`, and probably other programs as well. A constant (quick) memory increase happens mainly in the case of Molecular Dynamics (MD) simulations, some (huge) matrix diagonalization, finite field, numerical (first and second) derivations, and probably many others as well. 

The problem is solved in `OpenMPI` `4.0.7` and `4.1.2`. Version `4.0.3` is not affected. The patches were taken from https://github.com/open-mpi/ompi/pull/8828

### CODE REVIEW

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb [hunk 1]
```diff
@@ -14,13 +14,16 @@ sources = [SOURCELOWER_TAR_BZ2]
 patches = [
     'OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch',
     'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch'
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb:16]: @hajgato missing trailing `,`

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-intel-compilers-2021.2.0.eb [hunk 2]
```diff
@@ -14,13 +14,16 @@ sources = [SOURCELOWER_TAR_BZ2]
 patches = [
     'OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch',
     'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch'
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-intel-compilers-2021.2.0.eb:16]: @hajgato missing trailing `,`

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb (hunk 1 above)
Comment (hajgato) [reply easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-10.3.0.eb]: done

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-intel-compilers-2021.2.0.eb (hunk 2 above)
Comment (hajgato) [reply easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-intel-compilers-2021.2.0.eb]: done

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in total)
node2643.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/51198d5dfc86a80fe48bd2cad19cdece for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/11f3194d9120170c550773078ff22d37 for a full test report.

**Pull request merged**